src/inverted_index.c

```diff
@@ -76,9 +76,26 @@ void IndexReader_OnReopen(RedisModuleKey *k, void *privdata) {
 
   // If the key is valid, we just reset the reader's buffer reader to the current block pointer
   ir->idx = RedisModule_ModuleTypeGetValue(k);
-  size_t offset = ir->br.pos;
-  ir->br = NewBufferReader(IR_CURRENT_BLOCK(ir).data);
-  ir->br.pos = offset;
+
+  // the gc marker tells us if there is a chance the keys has undergone GC while we were asleep
+  if (ir->gcMarker == ir->idx->gcMarker) {
+    // no GC - we just go to the same offset we were at
+    size_t offset = ir->br.pos;
+    ir->br = NewBufferReader(IR_CURRENT_BLOCK(ir).data);
```

Comment:
Oh, it actually creates a bug, there was a reason for it! :) we have to do the `size_t offset = ir->br.pos;` thing before that.